File: 1317. Convert Integer to the Sum of Two No-Zero Integers/convert_integer_to_the_sum_of_two_no-zero_integers.cpp
// Approach - I
// T.C.->O(nlogn)
// S.C.->O(1)
class Solution {
public:
    vector<int> getNoZeroIntegers(int n) {
        for(int a=1;a<n;a++)
        {
            int b=n-a;
            if((to_string(a).find('0'))==string::npos && (to_string(b).find('0'))==string::npos)
            {
                return {a,b};
            }
        }
        return { };
    }
};


// Approach - II
class Solution {
public:
    int countZeros(long long n) {
    if (n == 0) return 1; // special case
    int count = 0;
    while (n > 0) {
        if (n % 10 == 0) count++;
        n /= 10;
    }
    return count;
  }
    vector<int> getNoZeroIntegers(int n) {
        vector<int> ans;
        for (int i = 1; i < n; i++) {
            if (countZeros(i) == 0 && countZeros(n-i)==0){
                ans = {i, n-i};
                break;
            }
        }
        return ans;
    }
};

File: 1317. Convert Integer to the Sum of Two No-Zero Integers/convert_integer_to_the_sum_of_two_no-zero_integers.py
# Approach - I
# T.C.->O(nlogn)
# S.C.->O(1)
class Solution:
    def getNoZeroIntegers(self, n: int) -> List[int]:
        for A in range(1, n):
            B = n - A
            if "0" not in str(A) + str(B):
                return [A, B]
        return []


# Approach - II
class Solution:
    def getNoZeroIntegers(self, n: int) -> List[int]:
        def check(val):
            while val:
                tmp=val%10
                if tmp==0:
                    return False
                val//=10
            return True
        for a in range(1,n): 
            b=n-a
            if check(a) and check(b):
                return [a,b]turn ans;
    }
};
